code/__HELPERS/view.dm

```diff
@@ -1,12 +1,7 @@
 /proc/getviewsize(view)
-	var/viewX
-	var/viewY
 	if(isnum(view))
-		var/totalviewrange = 1 + 2 * view
-		viewX = totalviewrange
-		viewY = totalviewrange
-	else
-		var/list/viewrangelist = splittext(view,"x")
-		viewX = text2num(viewrangelist[1])
-		viewY = text2num(viewrangelist[2])
-	return list(viewX, viewY)
\ No newline at end of file
+		. = 1 + 2 * view
+		return list(.,.)
+	if(istext(view))
```

Comment:
I suppose the istext makes a difference though

Reply:
| Proc Name | Self CPU | Total CPU | Real Time | Calls |
| --- | --- | --- | --- | --- |
| /proc/new_viewsize | 0.358 | 0.360 | 0.368 | 100000 |
| /proc/old_viewsize | 0.337 | 0.338 | 0.347 | 100000 |
| /proc/my_viewsize | 0.256 | 0.257 | 0.269 | 100000 |

This istext is the problem, it was just an else before.

Also this proc could be made faster anyway

```
/proc/getviewsize(view)
	if(length(view))
		return splittext(view, "x")
	else
		var/num = 1 + 2 * view
		return list(num, num)
```